### PR TITLE
LG-4144 | LG-4192: [CCP] TextArea

### PR DESCRIPTION
**Pull request by stephl3**
## ✍️ Proposed changes

`TextArea`
- `FormField` styling applies to `TextArea`
- passes aria props to `FormField`
- adds `successMessage` prop
- adds default state messages

🎟 _Jira ticket:_ [LG-4144](https://jira.mongodb.org/browse/LG-4144) | [LG-4192](https://jira.mongodb.org/browse/LG-4192)

### For bug fixes, new features & breaking changes

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### For new components

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README
- [ ] I have verified the Live Example looks as intended on the design website.

## 🧪 How to test changes

- Figma: https://www.figma.com/file/8M2FIDohVwZhC05jn0z17K/LG-2103%3A-Text-input-error-state?type=design&node-id=14674-86000&mode=design&t=RVhTpuWS4QhysQAO-4
- Review storybook and chromatic

### CODE REVIEW

File: .changeset/pretty-toes-tan.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+'@leafygreen-ui/text-area': major
+---
+
+[LG-4144](https://jira.mongodb.org/browse/LG-4144)
+
+1. `FormField` styling changes apply to `TextArea`. [See style changes here](https://github.com/mongodb/leafygreen-ui/blob/main/packages/form-field/CHANGELOG.md#102)
+
+2. A default `errorMessage` of `'This input needs your attention'` will render below text area when state is invalid.
```
Comment (TheSonOfThomp) [inline .changeset/pretty-toes-tan.md:9]: This feels strange to me. Where is this coming from?

File: .changeset/pretty-toes-tan.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+3. A default `successMessage` of `'Success'` will render when state is valid. `successMessage` prop accepts a custom string.
```
Comment (TheSonOfThomp) [inline .changeset/pretty-toes-tan.md:11]: `successMessage` didn't exist before did it?

**Review by TheSonOfThomp — COMMENTED** [review]
Just curious about the default `errorMessage`/`successMessage` values. Where are those coming from?

File: .changeset/pretty-toes-tan.md (hunk 1 above)
Comment (stephl3) [reply .changeset/pretty-toes-tan.md:9]: @TheSonOfThomp yes, this is a new spec from design so the error icon doesn't show up by itself

File: .changeset/pretty-toes-tan.md (hunk 2 above)
Comment (stephl3) [reply .changeset/pretty-toes-tan.md:11]: yes new spec from design

**Review by TheSonOfThomp — APPROVED** [review]

**Pull request merged**